sycl/test/extensions/annotated_usm/annotated_usm_negative.cpp

```diff
@@ -0,0 +1,87 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsyntax-only -Xclang -verify -Xclang -verify-ignore-unexpected=note %s
+
+// Expected failed tests for annotated USM allocation:
+// 1. Given properties contain invalid runtime property
+// 2. Given properties contain invalid compile-time property
+// 3. usm_kind in the property list conflicts with the function name
+// 4. required usm_kind is not provided in the property list
+
+#include <sycl/sycl.hpp>
+
+#include "fake_properties.hpp"
+
+#define TEST(f, args...)                                                       \
+  { auto ap = f(args); }
+
+using namespace sycl::ext::oneapi::experimental;
+using namespace sycl::ext::intel::experimental;
+using alloc = sycl::usm::alloc;
+
+constexpr int N = 10;
+
+// clang-format off
+
+void testInvalidRuntimeProperty(sycl::queue &q) {
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid runtime property in the property list.}}
```

Comment:
Can these error messages that state an invalid property was found in the property list, actually name the invalid property?
Otherwise, it could be difficult for the user to guess which of their properties were invalid.

Reply:
There are two errors emitted for each test case here:
- `Found invalid compile-time property in the property list.` directs user to the line of the annotated malloc function call
-  `Property is invalid for the given type.` points out the invalid property (which is omitted with regular expression `{{.+}}`)